<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TestingWindow</class>
 <widget class="QDialog" name="TestingWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>320</width>
    <height>381</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <widget class="QPushButton" name="sortingTest_1">
   <property name="geometry">
    <rect>
     <x>90</x>
     <y>80</y>
     <width>141</width>
     <height>32</height>
    </rect>
   </property>
   <property name="text">
    <string>Sort(unsorted)</string>
   </property>
   <property name="autoDefault">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QPushButton" name="sortingTest_2">
   <property name="geometry">
    <rect>
     <x>90</x>
     <y>110</y>
     <width>141</width>
     <height>32</height>
    </rect>
   </property>
   <property name="text">
    <string>Sort(partly sorted)</string>
   </property>
   <property name="autoDefault">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QPushButton" name="sortingTest_3">
   <property name="geometry">
    <rect>
     <x>90</x>
     <y>140</y>
     <width>141</width>
     <height>32</height>
    </rect>
   </property>
   <property name="text">
    <string>Sort(nearly sorted)</string>
   </property>
   <property name="autoDefault">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QPushButton" name="sortingTest_4">
   <property name="geometry">
    <rect>
     <x>90</x>
     <y>170</y>
     <width>141</width>
     <height>32</height>
    </rect>
   </property>
   <property name="text">
    <string>Sort(reversed)</string>
   </property>
   <property name="autoDefault">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QComboBox" name="sortTypeComboBox">
   <property name="geometry">
    <rect>
     <x>180</x>
     <y>20</y>
     <width>131</width>
     <height>21</height>
    </rect>
   </property>
   <item>
    <property name="text">
     <string>Bubble sort</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Shaker sort</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Comb sort</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Insertion sort</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Shellsort</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Tree sort</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Gnome sort</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Selection sort</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Heapsort</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Quick sort</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Merge sort</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Bucket sort</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>LSD sort</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>MSD sort</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Bitonic sort</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Timsort</string>
    </property>
   </item>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>20</y>
     <width>171</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>Choose sorting method ➙</string>
   </property>
  </widget>
  <widget class="QPushButton" name="returnButton">
   <property name="geometry">
    <rect>
     <x>210</x>
     <y>350</y>
     <width>111</width>
     <height>32</height>
    </rect>
   </property>
   <property name="text">
    <string>Return</string>
   </property>
   <property name="autoDefault">
    <bool>false</bool>
   </property>
   <property name="default">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>60</y>
     <width>171</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string> ⇩Choose a sorting test⇩</string>
   </property>
  </widget>
  <widget class="QPushButton" name="questionButton_1">
   <property name="geometry">
    <rect>
     <x>220</x>
     <y>80</y>
     <width>31</width>
     <height>32</height>
    </rect>
   </property>
   <property name="text">
    <string>?</string>
   </property>
  </widget>
  <widget class="QPushButton" name="questionButton_2">
   <property name="geometry">
    <rect>
     <x>220</x>
     <y>110</y>
     <width>31</width>
     <height>32</height>
    </rect>
   </property>
   <property name="text">
    <string>?</string>
   </property>
  </widget>
  <widget class="QPushButton" name="questionButton_3">
   <property name="geometry">
    <rect>
     <x>220</x>
     <y>140</y>
     <width>31</width>
     <height>32</height>
    </rect>
   </property>
   <property name="text">
    <string>?</string>
   </property>
  </widget>
  <widget class="QPushButton" name="questionButton_4">
   <property name="geometry">
    <rect>
     <x>220</x>
     <y>170</y>
     <width>31</width>
     <height>32</height>
    </rect>
   </property>
   <property name="text">
    <string>?</string>
   </property>
  </widget>
  <widget class="QPushButton" name="questionButton_5">
   <property name="geometry">
    <rect>
     <x>290</x>
     <y>220</y>
     <width>31</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>?</string>
   </property>
  </widget>
  <widget class="QPushButton" name="questionButton_6">
   <property name="geometry">
    <rect>
     <x>290</x>
     <y>250</y>
     <width>31</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>?</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_5">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>220</y>
     <width>121</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>Set an array size ➙</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_6">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>250</y>
     <width>191</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>Set a  modulus ➙</string>
   </property>
  </widget>
  <widget class="QWidget" name="layoutWidget">
   <property name="geometry">
    <rect>
     <x>180</x>
     <y>220</y>
     <width>111</width>
     <height>31</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QLabel" name="label_3">
      <property name="text">
       <string>Size:</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLineEdit" name="sizeLineEdit">
      <property name="text">
       <string>1000</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="layoutWidget">
   <property name="geometry">
    <rect>
     <x>180</x>
     <y>250</y>
     <width>111</width>
     <height>31</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <widget class="QLabel" name="label_4">
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Mod:</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLineEdit" name="modulusLineEdit">
      <property name="text">
       <string>100</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QLabel" name="label_7">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>280</y>
     <width>191</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>Set a sorted koefficient ➙</string>
   </property>
  </widget>
  <widget class="QPushButton" name="questionButton_7">
   <property name="geometry">
    <rect>
     <x>290</x>
     <y>280</y>
     <width>31</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>?</string>
   </property>
  </widget>
  <widget class="QWidget" name="layoutWidget_2">
   <property name="geometry">
    <rect>
     <x>180</x>
     <y>280</y>
     <width>111</width>
     <height>31</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_3">
    <item>
     <widget class="QLabel" name="label_8">
      <property name="font">
       <font>
        <pointsize>8</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Sorted:</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLineEdit" name="sortedKoeffLineEdit">
      <property name="text">
       <string>10</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QLabel" name="label_9">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>310</y>
     <width>191</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>Set a swaps koefficient ➙</string>
   </property>
  </widget>
  <widget class="QPushButton" name="questionButton_8">
   <property name="geometry">
    <rect>
     <x>290</x>
     <y>310</y>
     <width>31</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>?</string>
   </property>
  </widget>
  <widget class="QWidget" name="layoutWidget_3">
   <property name="geometry">
    <rect>
     <x>180</x>
     <y>310</y>
     <width>111</width>
     <height>31</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_4">
    <item>
     <widget class="QLabel" name="label_10">
      <property name="font">
       <font>
        <pointsize>8</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Swaps:</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLineEdit" name="swapsLineEdit">
      <property name="text">
       <string>10</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QLabel" name="label_11">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>90</y>
     <width>16</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>↪︎</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_12">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>120</y>
     <width>16</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>↪︎</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_13">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>150</y>
     <width>16</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>↪︎</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_14">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>180</y>
     <width>16</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>↪︎</string>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
